app/models/coin.rb

```diff
@@ -211,6 +226,68 @@ def sparkline
     end
   end
 
+  def token_metrics_data(metric_type)
+    return nil unless has_token_metrics?
+
+    Rails.cache.fetch("coins/#{id}/#{metric_type}", expires_in: 1.day, race_condition_ttl: 10.seconds) do
```

Comment:
Might also be worthwhile from a performance perspective to assign this result to an instance variable so that each call to individual metric methods (i.e. `exchange_supply_data`, `token_retention_rate_data`, etc.) doesn't need to go over the wire to fetch from Redis:

```
@token_metrics_data ||= Rails.cache.fetch...
```